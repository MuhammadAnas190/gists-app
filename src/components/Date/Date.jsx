import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { getFormattedDate } from '../../utils/helpers';

export const Date = ({ label, time }) => (
  <Wrapper>
    <span>
      {label}
      :
    </span>
    <span>{time ? getFormattedDate(time) : 'N/A'}</span>
  </Wrapper>
);
const Wrapper = styled.div`
  display: flex;
  gap: 0.5rem;
  justify-content: center;
  align-items: center;
  font-weight: 400;
  font-size: 13px;
`;

Date.propTypes = {
  label: PropTypes.string.isRequired,
  time: PropTypes.string,
};
Date.defaultProps = {
  time: '',
};

export default Date;
