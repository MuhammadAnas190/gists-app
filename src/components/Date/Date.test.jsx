import { render } from '@testing-library/react';
import { Date } from './Date';

describe('Date Component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Date time="2023-10-10T10:10:40Z" label="Created At" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
