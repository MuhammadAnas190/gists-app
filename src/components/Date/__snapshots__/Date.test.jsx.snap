// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-aXZVg ZcvCR"
  >
    <span>
      Created At:
    </span>
    <span>
      10/10/2023
    </span>
  </div>
</DocumentFragment>
`;
